airflow/providers/weaviate/hooks/weaviate.py

```diff
@@ -396,28 +399,46 @@ def batch_data(
             .. seealso:: `batch_config_params options <https://weaviate-python-client.readthedocs.io/en/v3.25.3/weaviate.batch.html#weaviate.batch.Batch.configure>`__
         :param vector_col: name of the column containing the vector.
         :param retry_attempts_per_object: number of time to try in case of failure before giving up.
+        :param tenant: The tenant to which the object will be added.
```

Comment:
For ingesting data, it should determine objects to insert, objects to delete and objects that remain unchanged for each chunk. This would simplify the logic.